testing/test_mark.py

```diff
@@ -927,3 +927,35 @@ def test_parameterset_for_parametrize_marks(testdir, mark):
 def test_parameterset_for_parametrize_bad_markname(testdir):
     with pytest.raises(pytest.UsageError):
         test_parameterset_for_parametrize_marks(testdir, 'bad')
+
+
+def test_mark_expressions_no_smear(testdir):
+    testdir.makepyfile("""
+        import pytest
+
+        class BaseTests(object):
+            def test_something(self):
+                pass
+
+        @pytest.mark.FOO
+        class TestFooClass(BaseTests):
+            pass
+
+        @pytest.mark.BAR
+        class TestBarClass(BaseTests):
+            pass
+    """)
+
+    reprec = testdir.inline_run("-m", 'FOO')
+    passed, skipped, failed = reprec.countoutcomes()
+    dlist = reprec.getcalls("pytest_deselected")
+    assert passed == 1
+    assert skipped == failed == 0
+    deselected_tests = dlist[0].items
+    assert len(deselected_tests) == 1
+
+    # keywords smear
```

Comment:
Is this the intended behavior or just demonstrated an unwanted behavior that we plan to fix in the future?

Reply:
this is intended behaviour - i hope to kill keywords in a breaking release ^^ but thats a few years down the line as its not clear how to replace them